src/library_syscall.js

```diff
@@ -866,7 +866,7 @@ var SyscallsLibrary = {
     var ptr;
     var allocated = false;
     if (fd === -1) {
-      ptr = _malloc(len);
+      ptr = _memalign(4096, len);
```

Comment:
Either way is fine, since we'll squash the commits anyhow (unless there is a reason to keep them separate, which is rare). In general it's easier to add a new commit since then the new changes are clearer.